src/zkevm_specs/evm/execution/dataCopy.py

```diff
@@ -0,0 +1,85 @@
+from typing import Optional, Sequence, Tuple, Union, List
+
+from ..instruction import Instruction, Transition
+from ..table import (
+    AccountFieldTag,
+    BlockContextFieldTag,
+    BytecodeFieldTag,
+    CallContextFieldTag,
+    FixedTableRow,
+    RWTableRow,
+    Tables,
+    FixedTableTag,
+    TxContextFieldTag,
+    RW,
+    RWTableTag,
+    TxLogFieldTag,
+    TxReceiptFieldTag,
+    CopyDataTypeTag,
+)
+from ...util import FQ, N_BYTES_MEMORY_ADDRESS, N_BYTES_MEMORY_SIZE, IdentityPerWordGas
+
+
+def dataCopy(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    call_data_offset = instruction.call_context_lookup(CallContextFieldTag.CallDataOffset, RW.Read)
+    call_data_length = instruction.call_context_lookup(CallContextFieldTag.CallDataLength, RW.Read)
+    return_data_offset = instruction.call_context_lookup(
+        CallContextFieldTag.ReturnDataOffset, RW.Read
+    )
+    return_data_length = instruction.call_context_lookup(
+        CallContextFieldTag.ReturnDataLength, RW.Read
+    )
+
+    opcall_call_id = instruction.curr.call_id
+    precompile_call_id = instruction.next.call_id
```

Comment:
After entering this step, the current `call_id` is already updated to `precompile_call_id`, and `opcall_call_id` should be read from `rw_table`.

```suggestion
    opcall_call_id = instruction.call_context_lookup(CallContextFieldTag.CallerId, RW.Read)
    precompile_call_id = instruction.curr.call_id
```